java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -872,6 +876,42 @@ public CompletableFuture<Long> zrangestore(
         return this.zrangestore(destination, source, rangeQuery, false);
     }
 
+    @Override
+    public CompletableFuture<String[]> zunion(
+            @NonNull WeightableKeys weightableKeys, @NonNull Aggregate aggregate) {
+        String[] arguments =
+                concatenateArrays(
+                        weightableKeys.toArgs(), new String[] {AGGREGATE_REDIS_API, aggregate.toString()});
+        return commandManager.submitNewCommand(
+                ZUnion, arguments, response -> castArray(handleArrayResponse(response), String.class));
+    }
+
+    @Override
+    public CompletableFuture<String[]> zunion(@NonNull WeightableKeys weightableKeys) {
+        return commandManager.submitNewCommand(
+                ZUnion,
+                weightableKeys.toArgs(),
+                response -> castArray(handleArrayResponse(response), String.class));
+    }
+
+    @Override
+    public CompletableFuture<Map<String, Double>> zunionWithScores(
+            @NonNull WeightableKeys weightableKeys, @NonNull Aggregate aggregate) {
+        String[] arguments =
+                concatenateArrays(
+                        weightableKeys.toArgs(),
+                        new String[] {AGGREGATE_REDIS_API, aggregate.toString(), WITH_SCORES_REDIS_API});
```

Comment:
and you need 2 `toArgs` methods or one with argument
```suggestion
                        aggregate.toArgs(true));
```

Reply:
I'm fine with the `WITH_SCORES_REDIS_API` staying here.